### PR TITLE
Add channel -K option to kill all Meterpreter channels

### PR DESCRIPTION
**Pull request by sjanusz-r7**
This small PR adds in a new command to Meterpreter that allows the end user to kill all channels at once.

## Verification

- [ ] Start `msfconsole`
- [ ] Acquire a Meterpreter session
- [ ] If on Linux, `execute -f bash -i` twice or more, and background using `CTRL + z`
- [ ] If on Windows, `execute -f cmd.exe -i -H` twice or more, and background using `CTRL + z`
- [ ] **Verify** `channel -K` kills all channels
- [ ] **Verify** `channel -l` lists no active channels

### CODE REVIEW

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb [hunk 1]
```diff
@@ -427,6 +430,15 @@ def cmd_channel(*args)
...
     when :write
       cmd_write(chan)
+    when :kill_all
+      if client.channels.empty?
+        print_line('No active channels.')
+        return
+      end
+
+      client.channels.each_pair do |id, channel|
+        cmd_close(id)
```
Comment (adfoster-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb:440]: Does this ever raise exceptions? 👀

Comment (adfoster-r7) [reply lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb:440]: Looks like there's the potential of exceptions being raised in the scenario of a race condition, i.e. the channel is closed just as the `channels -K` call is made

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb [hunk 2]
```diff
@@ -436,8 +436,12 @@ def cmd_channel(*args)
...
       end
 
-      client.channels.each_pair do |id, channel|
-        cmd_close(id)
+      begin
+        client.channels.each_pair do |id, channel|
+          cmd_close(id)
+        end
+      rescue ::StandardError
+        print_line('Failed when trying to kill a channel.')
```
Comment (adfoster-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb:444]: which channel?

Comment (adfoster-r7) [reply lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb:444]: Looks like this still has the same issue, i.e. if one individual channel kill fails - the remainder won't be closed

File: lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb [hunk 3]
```diff
@@ -427,6 +430,19 @@ def cmd_channel(*args)
...
+      if client.channels.empty?
+        print_line('No active channels.')
+        return
+      end
+
+      print_line('Killing all channels...')
+      client.channels.each_pair do |id, channel|
+        channel._close
+      rescue ::StandardError
+        print_line("Failed when trying to kill channel: #{id}")
```
Comment (adfoster-r7) [inline lib/rex/post/meterpreter/ui/console/command_dispatcher/core.rb:443]:
```suggestion
        print_error("Failed when trying to kill channel: #{id}")
```

**Review by adfoster-r7 — APPROVED** [review]

**Comment by dwelch-r7** [comment]
LGTM 
image

**Comment by dwelch-r7** [comment]
# Release Notes

Adds in a new command to Meterpreter that allows the end user to kill all channels at once

**Pull request merged**